### ISSUE TITLE
Catch ping timeouts

### ISSUE THREAD

**Issue by sam1919** [gf3/IRC-js#48]
When the connection is lost somehow, I get no event about this (I already catch *).

The irc-client shows that my bot is gone with a "ping timeout" message.

The bot itself doesn't recognize this.

(Additional question: Is there a workaround?)

**Comment by gf3** [comment]
Is this still an issue on `2.0.0-beta`?

**Comment by gf3** [comment]
Closing, assuming fixed.

**Comment by sam1919** [comment]
Yes, it's still an issue. I'm using 2.0.0-beta. My bot already catches \* but doesn't get info if the connection has timed out. Is there some timeout at all? How big is it? How do you find out that a connection has been lost?

My workaround is catching \* and let the bot remember the last time some event happened. So I could start a reconnect if the delay was too big.

But I'd really like to see irc-js handle this.

**Comment by sam1919** [comment]
Please reopen the bug.

**Comment by gf3** [comment]
Which server are you connecting to?

**Comment by sam1919** [comment]
My IRC server ist irc.mibbit.net.

**Comment by sam1919** [comment]
Just to explain this in more detail:

I run the bot from my home PC. My internet provider disconnects every night at about 3 to 5 am (Unfortunately I cannot change this). So the bot disconnects every day at night. But it isn't able (using irc-js) to detect that it has been disconnected.
